runners/console/index.ts

```diff
@@ -231,6 +243,28 @@ export class Console extends Runner {
         .fileContains(filepath, content);
     }
 
+    async assertCloneRepository(step: Step, command: Command, result: RunResult) {
+        let repository = command.parameters[1];
+        let repoName = repository.slice(repository.lastIndexOf("/"), -4);
+        let directorypath = path.join(this.getWorkingDirectory(), "devonfw", "workspaces", "main", command.parameters[0], repoName);
+        
+        let process = child_process.spawnSync("git status -s", { shell: true, cwd: directorypath });
+        if(process.status != 0) {
+            result.returnCode = process.status;
+        }
+
+        if(process.stdout.toString()) {
+            console.log("Current git status: " + process.stdout.toString());
+            result.returnCode = 1;
+        }
```

Comment:
Can we move this part in an own assertion? Or is this too special and only needed for this use case? What do you think @maybeec?
At least we should throw an exception with a custom error message. If we just set the return code to 1, the assertion noErrorCode will throw the error. And this assertion just log that the return code is not 0. So we have no information about the information we get from the git status command.

Reply:
You are right. Would be better to provide a proper assertion for that. Something like assertRepositoryToBeClean